### PR TITLE
Using modern cmake (version >= 3.9) allows to use the library as submodule

### PR DESCRIPTION
**Pull request by pablospe**
It is based on [https://github.com/pablospe/cmake-example-library](https://github.com/pablospe/cmake-example-library).

Another feature now it is possible to use the version number:
   ` project(PoseLib VERSION 1.0.0)`

The library will generated POSELIB_DEBUG if it is in Debug mode, it is possible to print the status of the library (version + building type), this is been used in the benchmark (in the end of the report):
    `poselib_print_version(void)`

### CODE REVIEW

File: PoseLib/poselib.h.in [hunk 1]
```diff
@@ -1,10 +1,26 @@
...
+
 //
 // Autogenerated lisf of #include <...>
 // based on the .h header files in PoseLib/ folder
 //
 @LIB_INCLUDES_STRING@
 
+
+// Print PoseLib version and buidling type
+void poselib_print_version(void) {
```
Comment (vlarsson) [inline PoseLib/poselib.h.in:15]: This should be in namespace, so

`pose_lib::print_version()`

potentially we could also return a string here instead and then we would not have to include <iostream>

File: benchmark/benchmark.cc [hunk 2]
```diff
@@ -169,6 +170,12 @@ void display_result(const std::vector<pose_lib::BenchmarkResult> &results) {
         print_runtime(runtime_ns);
         std::cout << "\n";
     }
+
```
Comment (vlarsson) [inline benchmark/benchmark.cc:173]: Maybe this should be printed first instead of at the end.

File: build_linux.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
```
Comment (vlarsson) [inline build_linux.sh:1]: I don't think we should have these build scripts. Potentially we could update the README with some more detailed instructions, but I think it's fine currently.

File: build_ninja.sh [hunk 4]
```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
```
Comment (vlarsson) [inline build_ninja.sh:2]: Remove

File: build_win.sh [hunk 5]
```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
```
Comment (vlarsson) [inline build_win.sh:2]: Remove

**Comment by pablospe** [comment]
This is related to issue #1

**Pull request merged**